### PR TITLE
doc: Add KSP section and mention it in what's new

### LINKED ISSUE #8730: Missing docs for KSP (Kotlin Symbol Processing API﻿) support
[PR introducing KSP did not introduced any docs about it](https://github.com/micronaut-projects/micronaut-core/pull/8462/files). 

Reference: 
[Kotlin Symbol Processing API﻿](https://kotlinlang.org/docs/ksp-overview.html)

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: src/main/docs/guide/toc.yml [hunk 1]
```diff
@@ -247,6 +247,10 @@ languageSupport:
   groovy: Micronaut for Groovy
   kotlin:
     title: Micronaut for Kotlin
+    kaptOrKsp: Kotlin support via KAPT or KSP
+    kapt: kapt compiler plugin
```
Comment (graemerocher) [inline src/main/docs/guide/toc.yml]: Expand to proper titles:

```
    kapt: Kotlin Annotation Processing (KAPT)
    ksp: Kotlin Symbol Processing (KSP)
```

File: src/main/docs/guide/introduction/whatsNew.adoc [hunk 2]
```diff
@@ -1,6 +1,10 @@
 //Micronaut {version} includes the following changes:
 == 4.0.0
 
+=== Kotlin applications via KSP
```
Comment (graemerocher) [inline src/main/docs/guide/introduction/whatsNew.adoc]: Change to:

```
=== Experimental Support for Kotlin Symbol Processing (KSP)
```

File: src/main/docs/guide/languageSupport/kotlin/kaptOrKsp.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Micronaut Framework has offered support for Kotlin via https://kotlinlang.org/docs/reference/kapt.html[Kapt].
+
+With version 4.0, Micronaut Framework supports Kotlin also via https://kotlinlang.org/docs/ksp-overview.html[Kotlin Symbol Processing (KSP) API].
+
+Please note that `kapt` is in maintenance mode. Micronaut Kotlin users should move to our KSP support.
+
+____
+kapt is in maintenance mode. We are keeping it up-to-date with recent Kotlin and Java releases but have no plans to implement new features.
```
Comment (graemerocher) [inline src/main/docs/guide/languageSupport/kotlin/kaptOrKsp.adoc:8]: Rephrase to:

```
Please note that KAPT is in maintenance mode. Micronaut 4 includes experimental support for KSP which Kotlin users should consider migrating in the future.
```

File: src/main/docs/guide/languageSupport/kotlin/ksp.adoc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+You can build Micronaut applications with Kotlin and https://kotlinlang.org/docs/ksp-overview.html[KSP]:
```
Comment (graemerocher) [inline src/main/docs/guide/languageSupport/kotlin/ksp.adoc:1]: should probably include a build example:

```groovy
plugins {
    id("org.jetbrains.kotlin.jvm") version "1.8.0"
    id("com.google.devtools.ksp") version "1.8.0-1.0.9"
    id("org.jetbrains.kotlin.plugin.allopen") version "1.8.0"
    id("io.micronaut.application")
}

dependencies {
    ksp("io.micronaut:micronaut-inject-kotlin")
    kspTest("io.micronaut:micronaut-inject-kotlin")
    ....
}
```

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: src/main/docs/guide/languageSupport/kotlin/kaptOrKsp.adoc (hunk 3 above)
Comment (sdelamo) [reply src/main/docs/guide/languageSupport/kotlin/kaptOrKsp.adoc:8]: Close via https://github.com/micronaut-projects/micronaut-core/pull/8786/commits/d788a01765b9f0558b7bf4534797537827f0783e

File: src/main/docs/guide/introduction/whatsNew.adoc (hunk 2 above)
Comment (sdelamo) [reply src/main/docs/guide/introduction/whatsNew.adoc]: close via https://github.com/micronaut-projects/micronaut-core/pull/8786/commits/13dd300dc4b77838c4ef980d933ab5bd759c513a

File: src/main/docs/guide/toc.yml (hunk 1 above)
Comment (sdelamo) [reply src/main/docs/guide/toc.yml]: close via https://github.com/micronaut-projects/micronaut-core/pull/8786/commits/3bcf6717a31b9a36e7012e9248d068842fa1abd3

**Comment by jameskleeh** [comment]
The docs should tell users why they should switch. Perhaps discuss the issues with kapt and later versions of java + the edge cases that come with translating Kotlin code to Java stubs (use of default in Kotlin comes to mind). See https://github.com/micronaut-projects/micronaut-core/issues?q=is%3Aissue+label%3A%22relates-to%3A+native-kotlin%22+

File: src/main/docs/guide/languageSupport/kotlin/ksp.adoc (hunk 4 above)
Comment (sdelamo) [reply src/main/docs/guide/languageSupport/kotlin/ksp.adoc:1]: see: https://github.com/micronaut-projects/micronaut-core/pull/8786/commits/a9ff12bd0c38ad295d5679948925176b6d69a476

**Comment by sdelamo** [comment]
> The docs should tell users why they should switch. Perhaps discuss the issues with kapt and later versions of java + the edge cases that come with translating Kotlin code to Java stubs (use of default in Kotlin comes to mind). See https://github.com/micronaut-projects/micronaut-core/issues?q=is%3Aissue+label%3A%22relates-to%3A+native-kotlin%22+

I added https://github.com/micronaut-projects/micronaut-core/pull/8786/commits/0ce7229175d4aa917e064fcb05808b7dd2557bff

**Comment by sdelamo** [comment]
@graemerocher can you review it again?

**Review by graemerocher — APPROVED** [review]

**Pull request merged**